fix(curriculum): remove media query requirement, general improvements (#46639)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/14-responsive-web-design-22/learn-css-flexbox-by-building-a-photo-gallery/6153908a366afb4d57185c8d.md

# --description--

Flexbox has a main and cross axis. The main axis is defined by `flex-direction`, which has four possible values:

> - `row` (default): content direction, depending on writing mode.
> - `row-reverse`: opposite to the content direction.
> - `column`: top to bottom.
> - `column-reverse`: bottom to top.
> 
> Try the different values to see how they affect the layout.
> 
> When you are done, set an explicit `flex-direction` of `row` on the `.gallery` element.

Reviewer: ```suggestion
- `row` (default): horizontal axis with flex items from left to right
- `row-reverse`: horizontal axis with flex items from right to left
- `column`: vertical axis with flex items from top to bottom
- `column-reverse`: vertical axis with flex items from bottom to top

Try the different values to see how they affect the layout. When you are done, set the `flex-direction` of your `.gallery` to `row-reverse`.
```

Not sure if this is any better, wonder if we should use `row-reverse` here so it's not the default (test would need changing).

Author: The direction is writing mode agnostic. We can't say left to right. I originally used the CSS tricks list but it used an abbreviation for the writing mode so I combined it with some of MDNs text.

Author: That is a very long sentence to put after the values though. Maybe we/I can think of a way to incorporate the writing mode information in the main text before the values list.

Also, I don't think you meant to change the direction to `row-reverse` as that isn't what the rest of the code is like.

Reviewer: Yea, the row-reverse was more of an idea than suggestion. I think it's a good idea since it's something other that the default - but we can leave it as the default for now.

Author: I just think doing it for no reason other than it not being the default is a little odd. I did initially try to think of a way to use a non-default value but then I would have had to create more steps. One idea was to start it out in a column before the wrapping had been applied and then switch it back to a row after.

The reverse properties are usually also discouraged from an accessibility standpoint as they visually change the order of the elements without it being reflected in the DOM.

# --description--

Flexbox has a main and cross axis. The main axis is defined by the `flex-direction` property, which has four possible values:

- `row` (default): horizontal axis with flex items from left to right
- `row-reverse`: horizontal axis with flex items from right to left
- `column`: vertical axis with flex items from top to bottom
- `column-reverse`: vertical axis with flex items from bottom to top

**Note**: The axes and directions will be different depending on the text direction. The values shown are for a left-to-right text direction.

> Try the different values to see how they affect the layout.

Reviewer 2: ```suggestion
```

Author: Why? I think it's worth encouraging the camper to try the values and see how they affect the layout.

Reviewer 2: Some aren't in the field for learning. For them, if they can see something that can be skipped, and test it at later date, they would.

 I am afraid, if they "see try something out", they won't.

Author: > Some aren't in the field for learning.

I don't really know what you mean by this?

> I am afraid, if they "see try something out", they won't.

If they don't, they don't. I don't see how that is relevant to encouraging it.

Reviewer 3: I'm with lasjorg on this - I'm a big fan of encouraging campers to experiment with things. It's a great way to learn.

> When you are done, set an explicit `flex-direction` of `row` on the `.gallery` element.

Reviewer 2: there is no point of setting `flex-direction: row`, it will be redundant code, and I think it's frowned upon to have redundant code.
 
```suggestion
set an explicit `flex-direction` of `row-reverse` on the `.gallery` element.
```

Author: It isn't so much redundant as it is explicit.

I agree that explicitly setting it to what is the default value is a little odd but I also think reversing the order for no reason and without giving a reason is as well.

I can see both sides of the argument but I really think we shouldn't use reverse (or `order`) unless there is a very good reason for it. It is frowned upon from an accessibility perspective.

Reviewer 3: > I agree that explicitly setting it to what is the default value is a little odd

Honestly, I think it's pretty normal. Explicitly setting a value ensures that it's not accidentally changed by browser defaults, or CSS libraries, and such.

Reviewer 2: > explicitly setting it to what is the default value is a little odd (but actually not that uncommon)

>  by browser defaults, or CSS libraries, and such.

I am convinced, should we mention why in the steps?

Author: @Sboonny I'd prefer not to do that. At least, not in this PR. I think we are already doing too much in this single PR.

Author: @naomi-lgbt True, I did edit the comment to say that it isn't that uncommon. But without explaining why you would want to set the default explicitly I can see us getting asked about it.

Reviewer 2: I will open issue about it after this pull request merge.

If I don't lose the note 😅

# --hints--

> Your `.gallery` selector should have a `flex-direction` property set to `row` as the value.

Reviewer 2: ```suggestion
Your `.gallery` selector should have a `flex-direction` property set to `row-reverse` as the value.
```

```js

> assert(new __helpers.CSSHelp(document).getStyle('.gallery')?.flexDirection === 'row');

Reviewer 2: ```suggestion
assert(new __helpers.CSSHelp(document).getStyle('.gallery')?.flexDirection === 'row-reverse');
```

# --description--

When working with flexbox you should think of it as having two axis. The main axis and the cross axis. The main axis is defined by `flex-direction`, which has four possible values:

- `row`
- `row-reverse`
- `column`
- `column-reverse`

When `flex-direction` is set to `row` or `row-reverse`, the main axis is the horizontal inline direction. When `flex-direction` is set to `column` or `column-reverse`, the main axis is the vertical block direction.

Default value for `flex-direction` is `row` when making an element a flex container.

---

You are encouraged to try out the different values and see how they affect the layout. When you are done, set an explicit `flex-direction` of `row` on the `.gallery` element.

> Then give the `.gallery` element a `max-width` of `1400px` and center it using an auto `margin` by setting `0 auto` as the `margin` value.

Author: I assume you are on a phone. Not sure if being able to see the editable region and all the text at the same time is really necessary. The instructions are at the bottom and you should be able to see them and the editable region.

In my opinion, we can't prioritize screen real estate that much higher than the subject we are teaching. But I agree, that step does have a lot of text, it already did. I can see if I can compress the information down, but it might take some time. But I won't remove as much as you are suggesting as it doesn't teach what the property does at all.

I might have to move the max-width and auto margin to a different step as well.

> Then give the `.gallery` element a `max-width` of `1400px` and center it using an auto `margin` by setting `0 auto` as the `margin` value.

Reviewer 2: > I assume you are on a phone. Not sure if being able to see the editable region and all the text at the same time is really necessary. The instructions are at the bottom and you should be able to see them and the editable region.

It was the laptop 😅

> In my opinion, we can't prioritize screen real estate that much higher than the subject we are teaching. But I agree, that step does have a lot of text, it already did. I can see if I can compress the information down, but it might take some time. But I won't remove as much as you are suggesting as it doesn't teach what the property does at all.

Sure, I though we were introducing concepts, not going in depth like the legacy RWD.

> Then give the `.gallery` element a `max-width` of `1400px` and center it using an auto `margin` by setting `0 auto` as the `margin` value.

Reviewer 2: I do like the idea of making new step, so people can focus on new concepts.

> Then give the `.gallery` element a `max-width` of `1400px` and center it using an auto `margin` by setting `0 auto` as the `margin` value.

Author: Should be a bit better now. I added the direction information to the value list and move the max-with and centering to step 16.
